src/platform/posix/tcp_ip_channel.c

```diff
@@ -217,10 +217,21 @@ void TcpIpChannel_register_callback(NetworkChannel *untyped_self,
   if (pthread_attr_init(&self->receive_thread_attr) < 0) {
     throw("pthread_attr_init failed");
   }
+/* TODO: RIOT posix-wrappers don't have pthread_attr_setstack yet */
+#ifdef __USE_XOPEN2K
```

Comment:
I needed to do these changed to the tcp_ip_channel because the RIOT POSIX wrappers don't support `pthread_attr_setstack`. I will try to fix that on RIOT ASAP but for now I added this `define` to support both cases.
It is needed to do it like this because `pthread_attr_setstackaddr` is deprecated. So we only want to use it if not avoidable.

Reply:
OK, we should figure it out, the reason I added it was because on Zephyr you have to do it that way...